src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/DefaultRazorProjectHost.cs

```diff
@@ -307,20 +308,34 @@ internal async Task OnProjectChanged(IProjectVersionedValue<IProjectSubscription
         
         private HostDocument[] GetCurrentDocuments(IProjectSubscriptionUpdate update)
         {
-            if (!update.CurrentState.TryGetValue(Rules.RazorGenerateWithTargetPath.SchemaName, out var rule))
+            IProjectRuleSnapshot rule = null;
+
+            var documents = new List<HostDocument>();
+            if (update.CurrentState.TryGetValue(Rules.RazorComponentWithTargetPath.SchemaName, out rule))
```

Comment:
This means that we can technically find components in the project system for a 2.X project. However, the code generation for 2.X will generate something obviously wrong. I think this is probably fine unless it becomes an easy scenario to get into.